lib/ratchet/html_parser.rb

```diff
@@ -0,0 +1,30 @@
+require 'ox'
+
+module Ratchet
+  class HtmlParser
+    def call(html)
+      document = Ox.parse(html)
+      parse(document)
+    end
+
+    private
+
+    def parse(element)
+      if element.is_a?(Ox::Element)
+        parse_element(element)
+      else
+        [:static, element]
+      end
+    end
+
+    def parse_element(element)
+      [
```

Comment:
In reviewing this PR I had a bit of trouble understanding this data structure—and similar ones, if there are more than one. That's part of my confusion: I wasn't 100% sure if this same structure type was used everywhere, or if different parts of the app used slightly different ones!

Is it something you defined for `ratchet-ruby`, or is it something related to Temple or Tilt? If something you defined, I wonder if it could increase maintainability to use a hash instead of an array. For example, I'm not clear on what `[:multi]` means. Then again, there could be a performance impact when parsing large files…